SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialTraceUniqueId.cpp

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialTraceUniqueId.h"
+
+using namespace SpatialGDK;
+
+FString EventTraceUniqueId::ToString() const
+{
+	return FString::Printf(TEXT("%0X"), Hash);
+}
+
+EventTraceUniqueId EventTraceUniqueId::GenerateForRPC(Worker_EntityId Entity, uint8 Type, uint64 Id)
+{
+	uint32 ComputedHash = HashCombine(HashCombine(GetTypeHash(static_cast<int64>(Entity)), GetTypeHash(Type)), GetTypeHash(Id));
```

Comment:
@improbable-danny wondering if adding an `Id` like this could solve the uniqueness problem for property updates. Couple of questions:

1. Is this `Id` unique across all workers too? Or just local to this worker?
2. Is the reason you didn't add an `Id` like this to the property update hash because this ID is already something which is sent with every RPC and therefore doesn't require you to send more data over the wire to use for RPCs, but you would have to attach such an ID to property update payloads?
3. Is this has actually sent over the wire? I don't think it is, right? If not, can we actually just ensure each of the inputs to the hash (in this case entity ID, type and ID) are logged as fields of the send and process RPC events and we don't even have to have this hash or worry about collisions?

Reply:
The Id alone isn't unique but it probably is combined with Entity and Type (type being either a server or client endpoint) and only one worker/client is authoritive over these respectively 

For properties the situation was slightly more complicated and required transmitting more data but we thought we could get away with it by using the shortest-path to the next property update (which also includes property updates being combined) 

These aren't sent over the wire, we can just log the inputs to the event instead though - was trying to do it in a generic-ish kind of way for all events but we could code this knowledge into the tool